### PR TITLE
changed: put CFileItemList in separate source file

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Split out CFileItemList from FileItem.cpp

## Motivation and context
Ease maintainability, smaller sources, one-class-per-file

## How has this been tested?
It builds.

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/view/GUIViewState.cpp
Comment (78andyp) [inline xbmc/view/GUIViewState.cpp:1]: Included twice

Comment (notspiff) [reply xbmc/view/GUIViewState.cpp:1]: fixed.

**Review by garbear — APPROVED** [review]
I imagine some of those `#include "FileItem.h"` can be removed, but that seems like a tedious task.

**Comment by notspiff** [comment]
in fact every single one of them single a CFileItemList is a CFileItem and thus includes FileItem.h but I don't rly see the gain and this keeps the door open for a day when that is not the case.

**Pull request merged**